### PR TITLE
Eng gogetu all

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to module CHANGELOG.md are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

**Review by benbp — APPROVED** [review]

**Comment by richardpark-msft** [comment]
@jhendrixMSFT, so basically a pre-PR step (for us) is to make sure we update all deps to the latest?

**Comment by jhendrixMSFT** [comment]
That's correct.

**Review by richardpark-msft — APPROVED** [review]
I can't think of any time we've purposefully tried to use an older version of a library so this seems safe...

**Comment by jhendrixMSFT** [comment]
Hmm ok I added support to skip this step in case it becomes a problem (which should be rare, if ever).

File: eng/pipelines/templates/variables/globals.yml [hunk 1]
```diff
@@ -12,5 +12,8 @@ variables:
   # Set the default value so that 'Build Performance Tests' step can resolve it
   UseAzcoreFromMain: $[eq(variables['Use.AzcoreFromMain'], 'true')]
 
+  # Set to disable analyze step for requiring latest version of dependencies
+  SkipGoGetUAll: $[eq(variables['Skip.GoGetUAll'], 'true')]
```
Comment (jhendrixMSFT) [inline eng/pipelines/templates/variables/globals.yml:16]: @benbp does this logic and above usage look correct?

**Pull request merged**